src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -149,7 +149,7 @@ const SignOrExecuteForm = ({
 
     // If txId isn't passed in props, it's a newly created tx
     // Redirect to the single tx view
-    if (redirectToTx && !txId) {
+    if ((redirectToTx && !txId) || isRejection) {
```

Comment:
Should be
`if (redirectToTx && (!txId || isRejection)) {

Reply:
We could possibly simplify the condition to `if (redirectToTx) ...` and just set the prop explicitly in those cases e.g. when using `SignOrExecuteForm` in `ConfirmProposedTx` we could pass `redirectToTx={false}` and could ignore the `txId` prop for the redirection.

Reply:
Good idea.